### PR TITLE
Fix the struct `KeyError` diagnostics

### PR DESCRIPTION
**Pull request by scottming**
We haven't handled the `KeyError` case before

```elixir
19:42:14.925 [error] Invalid diagnostic: {:exception, {%FunctionClauseError{module: LXical.RemoteControl.Build.Error, function: :error_to_diagnostic, arity: 4, kind: nil, args: [#LXical.Document<path: "/Users/scottming/Code/lexical/apps/remote_control/test/lexical/remote_control/build/error_test.exs", version: 60, dirty?: true, lines: %Lines<"defmodule Lexical.RemoteControl.Build.ErrorTest do..."(520 lines)>, ...>, %KeyError{key: :c, term: nil, message: "key :c not found"}, [{Foo, :__struct__, 1, [file: ~c"expanding struct
```

### CODE REVIEW

File: apps/remote_control/lib/lexical/remote_control/dispatch/handlers/indexing.ex [hunk 1]
```diff
@@ -11,7 +11,8 @@ defmodule Lexical.RemoteControl.Dispatch.Handlers.Indexing do
   use Dispatch.Handler, [file_quoted(), filesystem_event()]
 
   def on_event(file_quoted(document: document, quoted_ast: quoted_ast), state) do
-    reindex(document, quoted_ast)
+    {time, _} = :timer.tc(fn -> reindex(document, quoted_ast) end)
+    Logger.info("Reindexing #{document.path} took #{Lexical.Formats.time(time)}")
```
Comment (scohen) [inline apps/remote_control/lib/lexical/remote_control/dispatch/handlers/indexing.ex:15]: this seems unrelated.

Comment (scottming) [reply apps/remote_control/lib/lexical/remote_control/dispatch/handlers/indexing.ex]: ah, yes, deleted.

**Review by scohen — APPROVED** [review]
